packages/amplify-e2e-core/src/utils/rds.ts

```diff
@@ -126,6 +134,125 @@ export const createRDSInstance = async (
   }
 };
 
+/**
+ * Creates a new RDS Aurora serverless V2 cluster with one DB instance using the given input configuration.
+ * @param config Configuration of the database cluster. If password is not passed an RDS managed password will be created.
+ * @returns EndPoint address, port and database name of the created RDS cluster.
+ */
+export const createRDSCluster = async (
+  config: RDSConfig,
+): Promise<{
+  clusterArn: string;
+  endpoint: string;
+  port: number;
+  dbName: string;
+  password: string;
+  managedSecretArn: string;
+  dbInstance: DBInstance;
+}> => {
+  const rdsClient = new RDSClient({ region: config.region });
+  const initialDBName = 'defaultdb';
+
+  const params: CreateDBClusterMessage = {
+    /** input parameters */
+    EnableHttpEndpoint: true,
+    Engine: 'aurora-postgresql',
+    DatabaseName: initialDBName,
+    DBClusterIdentifier: config.identifier,
+    MasterUsername: config.username,
+    MasterUserPassword: config.password,
+    // use RDS managed password, then retrieve the password and store in all other credential store options
+    ManageMasterUserPassword: !config.password,
+    ServerlessV2ScalingConfiguration: {
+      MinCapacity: 4,
+      MaxCapacity: 10,
+    },
+  };
+
+  const command = new CreateDBClusterCommand(params);
+
+  const instanceParams: CreateDBInstanceCommandInput = {
+    DBInstanceClass: 'db.serverless',
+    DBInstanceIdentifier: createInstanceIdentifier(config.identifier),
+    Engine: 'aurora-postgresql',
+    DBClusterIdentifier: config.identifier,
+    PubliclyAccessible: config.publiclyAccessible ?? true,
+  };
+
+  const instanceCommand = new CreateDBInstanceCommand(instanceParams);
+
+  try {
+    const rdsResponse = await rdsClient.send(command);
+    const availableResponse = await waitUntilDBClusterAvailable(
+      {
+        maxWaitTime: 3600,
+        maxDelay: 120,
+        minDelay: 60,
+        client: rdsClient,
+      },
+      {
+        DBClusterIdentifier: config.identifier,
+      },
+    );
+
+    if (availableResponse.state !== 'SUCCESS') {
+      throw new Error('Error in creating a new RDS cluster.');
+    }
+
+    const dbCluster = availableResponse.reason.DBClusters[0];
+    if (!dbCluster) {
+      throw new Error('RDS cluster details are missing.');
+    }
+
+    const instanceResponse = await rdsClient.send(instanceCommand);
+    const availableInstanceResponse = await waitUntilDBInstanceAvailable(
+      {
+        maxWaitTime: 3600,
+        maxDelay: 120,
+        minDelay: 60,
+        client: rdsClient,
+      },
+      {
+        DBInstanceIdentifier: instanceParams.DBInstanceIdentifier,
+      },
+    );
+
+    if (availableInstanceResponse.state !== 'SUCCESS') {
+      throw new Error('Error in creating a new RDS instance inside the cluster.');
+    }
+
+    let password = config.password;
+    let masterUserSecret;
+
+    if (!config.password) {
```

Comment:
It looks like this method is only ever called from test invocations that use Data API. If that is the case, we should be using the RDS cluster master password ARN, and never returning the password.

If I'm mistaken, consider refactoring this function into 2 different functions for each use case, to minimize nested if/thens and make the return values purpose-built for the use case.